Xamarin.Forms.Platform.Android/AppCompat/ButtonRenderer.cs

```diff
@@ -292,11 +297,34 @@ class ButtonClickListener : Object, IOnClickListener
 
 			#endregion
 
-			public void OnClick(global::Android.Views.View v)
+			public void OnClick(AView v)
 			{
 				var renderer = v.Tag as ButtonRenderer;
 				((IButtonController)renderer?.Element)?.SendClicked();
 			}
 		}
+
+		class ButtonTouchListener : Object, IOnTouchListener
+		{
+			#region Statics
+
+			public static readonly Lazy<ButtonTouchListener> Instance = new Lazy<ButtonTouchListener>(() => new ButtonTouchListener());
+
+			#endregion
+
+			public bool OnTouch(AView v, AMotionEvent e)
+			{
+				var renderer = v.Tag as ButtonRenderer;
+				if (e.Action == AMotionEventActions.Down)
+				{
+					((IButtonController)renderer?.Element)?.SendPressed();
```

Comment:
Maybe I'm overthinking this. What happens if you touch down with one finger and touch down with another without lifting the first? Will this call `SendPressed()` twice? Same goes for released.